openedx/core/djangoapps/user_api/preferences/views.py

```diff
@@ -0,0 +1,208 @@
+"""
+NOTE: this API is WIP and has not yet been approved. Do not use this API without talking to Christina or Andy.
+
+For more information, see:
+https://openedx.atlassian.net/wiki/display/TNL/User+API
+"""
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework import permissions
+
+from django.utils.translation import ugettext as _
+
+from openedx.core.lib.api.parsers import MergePatchParser
+from ..errors import UserNotFound, UserNotAuthorized, PreferenceValidationError, PreferenceUpdateError
+from .api import (
+    get_user_preference, get_user_preferences, set_user_preference, update_user_preferences, delete_user_preference
+)
+
+
+class PreferencesView(APIView):
+    """
+        **Use Cases**
+
+            Get or update the user's preference information. Updates are only supported through merge patch.
+            Preference values of null in a patch request are treated as requests to remove the preference.
+
+        **Example Requests**:
+
+            GET /api/user/v0/preferences/{username}/
+
+            PATCH /api/user/v0/preferences/{username}/ with content_type "application/merge-patch+json"
+
+        **Response Value for GET**
+
+            A JSON dictionary will be returned with key/value pairs (all of type String).
+
+            If a user without "is_staff" access has requested preferences for a different user,
+            this method returns a 404.
+
+            If the specified username does not exist, this method returns a 404.
+
+        **Response for PATCH**
+
+            Users can only modify their own preferences. If the requesting user does not have username
+            "username", this method will return with a status of 404.
+
+            This method will also return a 404 if no user exists with username "username".
+
+            If "application/merge-patch+json" is not the specified content_type, this method returns a 415 status.
+
+            If the update could not be completed due to validation errors, this method returns a 400 with all
+            preference-specific error messages in the "preference_errors" field of the returned JSON.
+
+            If the update could not be completed due to failure at the time of update, this method returns a 400 with
+            specific errors in the returned JSON.
+
+            If the update is successful, a 204 status is returned with no additional content.
+
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
```

Comment:
Just spoke with @andy-armstrong. I'd prefer letting inactive users interact with their settings, but do whatever is lowest effort if there is a large difference.